src/main.rs

```diff
@@ -0,0 +1,206 @@
+extern crate clap;
+extern crate unicode_normalization;
+
+use clap::{App, Arg};
+use patricia_tree::PatriciaSet;
+use std::cmp::Reverse;
+use std::collections::hash_map::DefaultHasher;
+use std::collections::{HashMap, HashSet};
+use std::fs::{metadata, File};
+use std::hash::{Hash, Hasher};
+use std::io::Read;
+use std::str;
+use unicode_normalization::UnicodeNormalization;
+
+const SPACES_FACTOR: usize = 6;
+
+fn get_raw_dict(name: String, resource_dir: &str) -> Vec<u8> {
+    let dict_path_elts = vec![resource_dir.to_string(), name];
+    let dict_path = dict_path_elts.join("/");
```

Comment:
```suggestion
    let dict_path: PathBuf = [resource_dir, name].iter().collect();
```
Use `Path` and `PathBuf` for paths. It's simpler and platform-agnostic.